src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -542,33 +542,17 @@ void ImageReplayer<I>::on_start_fail_finish(int r)
   Context *on_stop_finish(nullptr);
   {
     Mutex::Locker locker(m_lock);
-    if (m_stop_requested) {
-      assert(r == -EINTR);
-      dout(20) << "start interrupted" << dendl;
-      m_state = STATE_STOPPED;
-      m_stop_requested = false;
+    if (r < 0) {
+      derr << "start failed: " << cpp_strerror(r) << dendl;
     } else {
-      assert(m_state == STATE_STARTING);
-      dout(20) << "start failed" << dendl;
-      m_state = (r < 0) ? STATE_UNINITIALIZED : STATE_STOPPED;
+      dout(20) << "start interrupted" << dendl;
```

Comment:
1) run `unittest_rbd_mirror  --gtest_filter=TestImageReplayer.StartInterrupted` with debug log level. Now we have in logs:
```
...::on_start_fail_finish: start failed: (4) Interrupted system call
```
Previously we had:
```
...::on_start_fail_finish: start interrupted
```
2) Run rbd_mirror.sh test. Look at the log for the secondary mirror. You will see
```
...::on_start_fail_finish: start interrupted
```
periodically. Previously we did not have these messages here.